File: How to write an SFTP file transfer client for WinForms/[C#]-How to write an SFTP file transfer client for WinForms/C#/ClientController_Fields.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace ClientSample
{
    partial class ClientController
    {
        private bool _reconnecting; // Indicates whether the connection is closed unexpectedly and the app is about to reconnect.

        private ConnectionState _state; // The connection state.
        private ServerProtocol _serverType;

        private string _currentDirectory;
        string _currentLocalDirectory;
        private bool _aborting;

        /// <summary>
        /// Gets the current local directory.
        /// </summary>
        public string CurrentLocalDirectory
        {
            get { return _currentLocalDirectory; }
        }

        /// <summary>
        /// Gets the current remote directory.
        /// </summary>
        public string CurrentRemoteDirectory
        {
            get { return _currentDirectory; }
        }
    }
}

File: How to write an SFTP file transfer client for WinForms/[VB]-How to write an SFTP file transfer client for WinForms/VB/ClientController_Fields.vb
Imports System.Text
Imports System.Threading

Namespace ClientSample
	Partial Friend Class ClientController
		Private _reconnecting As Boolean ' Indicates whether the connection is closed unexpectedly and the app is about to reconnect.

		Private _state As ConnectionState ' The connection state.
		Private _serverType As ServerProtocol

		Private _currentDirectory As String
		Private _currentLocalDirectory As String
		Private _aborting As Boolean

		''' <summary>
		''' Gets the current local directory.
		''' </summary>
		Public ReadOnly Property CurrentLocalDirectory() As String
			Get
				Return _currentLocalDirectory
			End Get
		End Property

		''' <summary>
		''' Gets the current remote directory.
		''' </summary>
		Public ReadOnly Property CurrentRemoteDirectory() As String
			Get
				Return _currentDirectory
			End Get
		End Property
	End Class
End Namespace
